### ISSUE TITLE
Options

### ISSUE THREAD

**Issue by Sugarbank** [rtqichen/style-swap#5]
Hey dude!
I'm loving playing with your implementation of style transfer.It's very cool!
Just wondering if you could explain some of the parameters a bit more?

--pooling               (default 'avg')         One of [avg|max]
--numSwap               (default 1)             Number of times to perform the style swap operation [Higher = More Style Contrast]

--decoder               (default '')            Path to a trained decoder
--optim                                         If set, decoder is only used for initialization and optimization still occurs

--learningRate          (default 0.05)          Learning rate for optimisation

**Comment by rtqichen** [comment]
Glad you're liking it.

--pooling changes the pooling layers of the pertained VGG network to use
either average pooling or max pooling. The average tends to give smoother
results.

--numSwap [int] allows performing the style swap operation multiple times.
This takes the result image and uses it as content in more swaps.

--decoder [path] takes a path to a trained inverse network. I should rename
this.

--optim is a flag that lets you use an inverse network first and then
continue to use optimization. This ideally is faster than using just
optimization while getting the same results.

--learningRate is the learning rate for Adam optimizer. You probably don't
need to tweak this.

On Jan 1, 2017 03:48, "Sugarbank" <notifications@github.com> wrote:

Hey dude!
I'm loving playing with your implementation of style transfer.It's very
cool!
Just wondering if you could explain some of the parameters a bit more?

--pooling (default 'avg') One of [avg|max]
--numSwap (default 1) Number of times to perform the style swap operation
[Higher = More Style Contrast]

--decoder (default '') Path to a trained decoder
--optim If set, decoder is only used for initialization and optimization
still occurs

--learningRate (default 0.05) Learning rate for optimisation

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/rtqichen/style-swap/issues/5>, or mute the thread

**Comment by Sugarbank** [comment]
Thanks so much for sending thru these explanations:)
How tricky do you think it would be to try and corporate optical flow into your implementation?

https://github.com/manuelruder/artistic-videos






Christian J Heinrich  | Direction/Design/Motion



[cid:56EB7795-138F-455C-87AF-D26619CBA0AE@gateway]


M +61 (0) 422 247 152

E. sugarbank@hotmail.com<mailto:sugarbank@hotmail.com>
R: vimeo.com/SUGARBANK<http://vimeo.com/SUGARBANK/reel>




On 2 Jan 2017, at 4:01 AM, Ricky Chen <notifications@github.com<mailto:notifications@github.com>> wrote:

Glad you're liking it.

--pooling changes the pooling layers of the pertained VGG network to use
either average pooling or max pooling. The average tends to give smoother
results.

--numSwap [int] allows performing the style swap operation multiple times.
This takes the result image and uses it as content in more swaps.

--decoder [path] takes a path to a trained inverse network. I should rename
this.

--optim is a flag that lets you use an inverse network first and then
continue to use optimization. This ideally is faster than using just
optimization while getting the same results.

--learningRate is the learning rate for Adam optimizer. You probably don't
need to tweak this.

On Jan 1, 2017 03:48, "Sugarbank" <notifications@github.com<mailto:notifications@github.com>> wrote:

Hey dude!
I'm loving playing with your implementation of style transfer.It's very
cool!
Just wondering if you could explain some of the parameters a bit more?

--pooling (default 'avg') One of [avg|max]
--numSwap (default 1) Number of times to perform the style swap operation
[Higher = More Style Contrast]

--decoder (default '') Path to a trained decoder
--optim If set, decoder is only used for initialization and optimization
still occurs

--learningRate (default 0.05) Learning rate for optimisation

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/rtqichen/style-swap/issues/5>, or mute the thread

**Comment by Sugarbank** [comment]
Yeah I see what you mean.
Maybe if you did a 3d render and rendered the Optical flow as a 3d pass then used that but generating the optical flow data is slow for sure.
I’m really impressed by this implementation as it allows for custom image inout and actually quite a lot of parameters to play with.

I’m trying to work out how I can get an HD image out of it but seems I don’t have a beauty enough graphics card. I’m pretty newbie to programming and more of a gfx,Vfx person but maybe if I do some reseat.You think it’s possible to tweak your implementation even more for faster speeds and higher resolutions?If so I’ll try and look into it.

Do you know of any other implementations that allow for custom style input as well?Seems most just rely on pre-tained styles for speed…..




Christian J Heinrich  | Direction/Design/Motion



[cid:56EB7795-138F-455C-87AF-D26619CBA0AE@gateway]


M +61 (0) 422 247 152

E. sugarbank@hotmail.com<mailto:sugarbank@hotmail.com>
R: vimeo.com/SUGARBANK<http://vimeo.com/SUGARBANK/reel>




On 5 Jan 2017, at 4:33 PM, Ricky Chen <notifications@github.com<mailto:notifications@github.com>> wrote:


Pretty tricky. Optical flow and style swap both share the same role as deciding where to place textures. It might be possible, but definitely not any time soon.. On the other hand, optical flow is very slow and pretty much defeats the purpose of this method.

**Comment by rtqichen** [comment]
If you want a high resolution image, then set --styleSize to be really low, or use a higher --patchStride (you get cooler results this way anyway). The main bottleneck is in the number of style patches, which is determined by these two parameters. Admittedly, this method is quite a memory hog.

I'm sure there will be better options in the future. But right now this is only method that allows custom style and cares about speed.

**Comment by Sugarbank** [comment]
Hi,
Just checking to see how you set the —optim flag…..is it 1 and 0?

**Comment by rtqichen** [comment]
No need to specify a number. It's just a flag.

**Comment by Sugarbank** [comment]
Hi,
I’m getting quite a difference between frames when batching a folder of stills( that i’ve derived from a quicktime.)
The process isn’t do the stills in order…Could this be the issue?
Or is there a setting which adds temporal smoothness or a seed function.

Also  would it be possible to reduce the size of the VGG to save on memory usage for processing images at a higher resolution?

Cheers
